pennylane/ops/op_math/controlled_class.py

```diff
@@ -310,6 +310,10 @@ def eigvals(self):
     def diagonalizing_gates(self):
         return self.base.diagonalizing_gates()
 
+    @property
+    def has_decomposition(self):
+        return not all(self.control_values)
```

Comment:
```suggestion
        if not all(self.control_values):
            return True
        if hasattr(self.base, "_controlled"):
            return True
        return self.base.has_decomposition
```

Reply:
I think the conditions are even a bit more fine grained, but maybe I'm overlooking something here? :thinking: 
For example, if there are multiple control wires, having the `_controlled` method is not sufficient, right?

Reply:
Made a suggestion based on the structure of `decomposition` :)